### PR TITLE
remove 0.10 and 0.12 from supportedTargets

### PR DESCRIPTION
**Pull request by ralphtheninja**

### CODE REVIEW

**Comment by ralphtheninja** [comment]
@lgeiger Should be new major I guess? (Since tests broke)

File: index.js [hunk 1]
```diff
@@ -63,8 +63,6 @@ function getTarget (abi, runtime) {
 }
 
 var supportedTargets = [
-  {runtime: 'node', target: '0.10.48', abi: '11', lts: false},
-  {runtime: 'node', target: '0.12.17', abi: '14', lts: false},
```
Comment (juliangruber) [inline index.js]: should we add those to `deprecatedTargets`?

**Comment by ralphtheninja** [comment]
Solves https://github.com/mafintosh/prebuild/issues/176 (but need to update major version in dependencies first)

File: index.js (hunk 1 above)
Comment (ralphtheninja) [inline index.js]: What does the `deprecatedTargets` really do?

Comment (ralphtheninja) [inline index.js]: Hmm, if we add these to deprecatedTargets as well, then wouldn't we have to deprecate all other 0.10 and 0.12 versions as well? It's essentially the range 0.10.x and 0.12.x we want to deprecate, right?

Comment (juliangruber) [inline index.js]: when a version is in `deprecatedTargets`, this library will find it when resolving a specific `abi`/`target`. It won't however be used for prebuild etc, as that uses `supportedTargets`

Comment (juliangruber) [inline index.js]: This deprecates the whole range :)

Comment (ralphtheninja) [inline index.js]: > This deprecates the whole range :)

:+1: I'll update

Comment (ralphtheninja) [inline index.js]: Todeloo

**Comment by ralphtheninja** [comment]
@juliangruber ping

File: test/index.js [hunk 2]
```diff
@@ -5,8 +5,6 @@ var getTarget = require('../index').getTarget
 test('getTarget calculates correct Node target', function (t) {
   t.equal(getTarget(undefined), process.versions.node)
   t.equal(getTarget(null), process.versions.node)
-  t.equal(getTarget('11'), '0.10.48')
-  t.equal(getTarget('14'), '0.12.17')
```
Comment (juliangruber) [inline test/index.js]: those can be left here still :)

**Comment by lgeiger** [comment]
Thanks for deprecating Node 0.10 and 0.12 👍

File: test/index.js (hunk 2 above)
Comment (ralphtheninja) [inline test/index.js]: Doh of course! Sorry :)

**Comment by lgeiger** [comment]
Should we release it as a major or minor version?

**Comment by ralphtheninja** [comment]
That's a really good question. If you're paranoid, use major, but then prebuild needs to be updated, which is cool.

**Comment by ralphtheninja** [comment]
But then maybe prebuild also should be major? Gah, these things are tricky when it's not the api that has changed.

**Comment by mathiask88** [comment]
Yes, always tricky :) I think its a major because it changes behavior. I build my prebuilds with prebuild-ci and this PR would stop building 0.x and I wouldn't notice. On the other side, I don't like it as major, because every time a version drops we need a major release in prebuild. Maybe just state in the prebuild docs that supported targets may change over time.

**Comment by ralphtheninja** [comment]
@mathiask88 Make a PR! ;)

**Comment by mathiask88** [comment]
Oops, I was not right with `prebuild-ci`, it relies on the `--target` argument. But nevertheless I think it should definitely be patch for `node-abi` because it just updates the underlying data and not the API. `supportedTargets()` etc. is like `getTime()` u know what you want, but the data may change over time :)
And for `prebuild` it should be fine to state that in the readme. I can make a PR this week.

**Comment by mafintosh** [comment]
minor, not patch

**Comment by mafintosh** [comment]
And not major. Thats why we have allTargets.

**Comment by lgeiger** [comment]
Thanks for participating in this discussion, those things are always tricky!

I'll ship a new minor version. Yay for maintainability 🚀 

> And for prebuild it should be fine to state that in the readme. I can make a PR this week.

Awesome 👍

**Pull request merged**